ext/SymbolicsGroebnerExt.jl

```diff
@@ -70,4 +70,97 @@ function Symbolics.is_groebner_basis(polynomials::Vector{Num}; kwargs...)
     Groebner.isgroebner(polynoms; kwargs...)
 end
 
+function Symbolics.solve_multivar(eqs::Vector{Num}, vars::Vector{Num}, mult=false)
+    # do the trick
+    new_var = (@variables HAT)[1]
+
+    old_len = length(vars)
+    push!(vars, new_var)
+
+    new_eqs = []
+    generating = true
+
+    while generating
+        new_eqs = deepcopy(eqs)
+        eq = new_var
+        for i = 1:(old_len)
+            eq -= rand(1:10)*vars[i]
+        end
+        push!(new_eqs, eq)
+        new_eqs = convert(Vector{Any}, Symbolics.groebner_basis(new_eqs, ordering=Lex(vars)))
+
+        if length(new_eqs) <= length(vars) 
+            generating &= false
+        end
+
+        for i  in eachindex(new_eqs)[2:end]
+            generating |= all(Symbolics.degree(var) > 1 for var in Symbolics.get_variables(new_eqs[i]))
+        end
+    end
```

Comment:
It is possible to construct an example where this does not terminate. 
A more robust version could be to increase the range in `rand(1:10)` after each iteration.

Example.
Consider the case with two variables $x, y$ and the coefficients sampled by `rand(1:2)`. There are 4 choices for $HAT$:
- $HAT = x + y$
- $HAT = x + 2y$
- $HAT = 2x + y$
- $HAT = 2x + 2y$

Let the input system be $y (y - 1) = x (x - y) (x - 2y) (2x - y) = 0$.
The number of solutions is finite since the GB contains 2 polynomials with leading terms $y^3$ and $x^4$, respectively.

Notice however that none of the 4 choices for HAT would produce a basis of length 3 .

Reply:
```julia
julia> using Nemo, Groebner

julia> R, (x,y,H) = QQ["x","y","H"]
(Multivariate polynomial ring in 3 variables over QQ, QQMPolyRingElem[x, y, H])

# the GB looks nice
julia> groebner([y*(y - 1)*(y - 2), x*(x - y)*(x - 2y)*(2x - y)])
2-element Vector{QQMPolyRingElem}:
 y^3 - 3*y^2 + 2*y
 x^4 - 7//2*x^3*y + 7//2*x^2*y^2 - 3*x*y^2 + 2*x*y

# add H = i*x + j*y to the equations
julia> for i in 1:2
                  for j in 1:2
                      @info "" groebner([
                          y*(y - 1)*(y - 2),
                          x*(x - y)*(x - 2y)*(2x - y),
                          H - (i*x + j*y)
                      ])
                  end
              end
```

Reply:
Fix:
```julia
    n_iterations = 1
    
    while generating
        new_eqs = deepcopy(eqs)
        eq = new_var
        for i = 1:(old_len)
            eq -= rand(1:n_iterations*10)*vars[i]
        end
        push!(new_eqs, eq)
        new_eqs = convert(Vector{Any}, Symbolics.groebner_basis(new_eqs, ordering=Lex(vars)))

        if length(new_eqs) <= length(vars) 
            generating &= false
        end

        for i  in eachindex(new_eqs)[2:end]
            generating |= all(Symbolics.degree(var) > 1 for var in Symbolics.get_variables(new_eqs[i]))
        end
        
        n_iterations += 1
    end
```